app/src/main/java/com/example/taskyy/ui/viewmodels/AgendaViewModel.kt

```diff
@@ -53,53 +66,82 @@ class AgendaViewModel @Inject constructor(
         val monthFormatter = DateTimeFormatter.ofPattern("MMMM", Locale.ENGLISH)
         val dayOfTheMonthFormatter = DateTimeFormatter.ofPattern("d", Locale.ENGLISH)
         val dayOfTheWeekFormatter = DateTimeFormatter.ofPattern("E", Locale.ENGLISH)
+        val dateStringFormatter = DateTimeFormatter.ofPattern("d MMMM uuuu", Locale.ENGLISH)
 
-        val tempDateList: ArrayList<LocalDateTime> = java.util.ArrayList()
-        val dayList: ArrayList<Day> = ArrayList()
-
-        val dateList: ArrayList<LocalDateTime> = incrementDate(localDateTime, tempDateList)
+        state = state.copy(dateString = dateStringFormatter.format(localDateTime.plusDays(1)))
 
-        for (date in dateList) {
-            val day = Day(
+        val days = (1..6).map {
+            val date = localDateTime.plusDays(it.toLong())
+            Day(
+                dayOfTheMonth = dayOfTheMonthFormatter.format(date),
                 dayOfTheWeek = dayOfTheWeekFormatter.format(date),
-                dayOfTheMonth = dayOfTheMonthFormatter.format(date)
+                index = it,
+                date = dateStringFormatter.format(date)
             )
-            dayList.add(day)
         }
 
-
         state = state.copy(
             selectedMonth = monthFormatter.format(localDateTime).uppercase(),
-            selectedDay = dayList,
+            selectedDayList = days,
         )
     }
 
-    private fun incrementDate(
-        localDateTime: LocalDateTime,
-        dateList: ArrayList<LocalDateTime>,
-    ): ArrayList<LocalDateTime> {
-        for (i in 0..6) {
-            dateList.add(localDateTime.plusDays(i.toLong()))
-        }
-        return dateList
-    }
-
     private fun logout() {
         viewModelScope.launch {
             state = state.copy(isUserLoggingOut = true)
             state = state.copy(wasLogoutSuccessful = logoutUseCase.logout())
             state = state.copy(isUserLoggingOut = false)
+
+            if (state.wasLogoutSuccessful) {
+                eventChannel.send(AgendaEvent.LogoutSuccessful)
+            }
+        }
+    }
+
+    fun setUserInitials(email: String) {
+        viewModelScope.launch {
+            val name = agendaRepository.getUserName(email)
+            state = state.copy(name = name)
+            state = state.copy(initials = name
+                .split(' ')
+                .mapNotNull { it.firstOrNull()?.toString() }
+                .reduce { acc, s -> acc + s })
```

Comment:
If there's a middle name, you'll get 3 initials here. In that case, the expected behavior would probably taking the initials of the first and last name ignoring the middle name